env/build.go

```diff
@@ -57,8 +58,12 @@ func NewBuildEnv(environ []string, platform Platform, buildpack Buildpack) *Env
 // NewDetectEnv returns an detect-time Env from the given environment.
 //
 // Only keys in the BuildEnvIncludelist will be added to the Environment.
-func NewDetectEnv(environ []string) *Env {
```

Comment:
Now that `NewDetectEnv` and `NewBuildEnv` don't behave any differently, we could just remove `NewDetectEnv`. It didn't exist before https://github.com/buildpacks/lifecycle/pull/591

Reply:
This unfortunately introduces complexity in that we'll need to also pass a `buildpack` when constructing the env. Here's where we do it for build: https://github.com/buildpacks/lifecycle/pull/591/files?file-filters%5B%5D=.go&file-filters%5B%5D=.yml#diff-5dce70063c4b1b07e611b3fcd32da0445a0b95970028c8992beb9c955e38320fR76

This is unfortunate because the `DetectConfig` (which includes the env) is currently defined on the `Detector` (at the level of all buildpacks, see [here](https://github.com/buildpacks/lifecycle/blob/cf2d0ed8f2059b7c528b75b7da129cbe95c98618/detector.go#L119)) and now it would need to be per-buildpack. For build, I ended up changing the interface to accept the env: https://github.com/buildpacks/lifecycle/pull/591/files?file-filters%5B%5D=.go&file-filters%5B%5D=.yml#diff-e348e4b0ed5d889979d2fb076558d56656b133c84892137fdd110002e8948ec8R63 - but this is made easier by the fact that the logic to "clear" the env already lives in the `buildpack` package (see [here](https://github.com/buildpacks/lifecycle/blob/cf2d0ed8f2059b7c528b75b7da129cbe95c98618/buildpack/build.go#L191-L198)) whereas for detect it lives in the `cmd` package (see [here](https://github.com/buildpacks/lifecycle/blob/cf2d0ed8f2059b7c528b75b7da129cbe95c98618/cmd/lifecycle/detector.go#L98-L99)). 🤷🏼

Reply:
Fair enough. I just wanted to know how badly we wanted it 🙂

Reply:
Well, that's a good question. I think if I had been more vociferous in raising that question when working on the build phase, we might've concluded that the extra effort wasn't worth it. But it strikes me as a little silly if the detect and build phases worked differently in this respect. Since we have a larger goal of removing logic from `cmd` I would say let's proceed, but let me know if you feel differently.

Reply:
The extra effort is worth it. I was just being lazy.

Removed in latest commit.